UnityProject/Assets/Editor/Tests/ObstacleGeneratorTests.cs

```diff
@@ -0,0 +1,22 @@
+using UnityEngine;
+using NUnit.Framework;
+using MapFeatures.Obstacles;
+
+namespace AIMMOUnityTest
+{
+    [TestFixture]
+    internal class ObstacleGeneratorTests
+    {
+        [Test]
+        public void TestAdjustedLocation()
```

Comment:
There's many more edge cases we need to cover here imo.

Reply:
agree - what does AdjustedLocation mean? It seems that there could be a lot of location related tests?